src/git.ts

```diff
@@ -122,10 +122,36 @@ function httpGitRequest(
   return ServerConnection.makeRequest(fullUrl, fullRequest, setting);
 }
 
+/**
+ * Structure for the result of the Git Clone API.
+ */
+export interface GitCloneResult {
+  code: number;
+  message?: string;
+}
+
 /** Parent class for all API requests */
 export class Git {
   constructor() {}
 
+    /** Placeholder for the Git Clone API. */
+    async clone(path: string, url: string): Promise<GitCloneResult> {
+        try {
+            let response = await httpGitRequest('/git/clone', 'POST', {
+                current_path: path,
+                clone_url: url
+            });
+            if (response.status !== 200) {
+                return response.text().then(data => {
```

Comment:
This function is already `async`, I think this would be cleaner as `const data = await response.text()`

Reply:
Fair point -  This was a pattern in the other Git methods as well; I refactored some of them along with this.